### PR TITLE
Bug 1609734 - Swift: Adds use of overflow metric when pre-init queue overflows

### PR DESCRIPTION
**Pull request by travis79**

### CODE REVIEW

**Comment by travis79** [comment]
@brizental wins the reviewer roulette!  I can walk you through any of this you may have questions about, if that makes it more comfortable for you to review.

File: glean-core/ios/Glean/Dispatchers.swift [hunk 1]
```diff
@@ -176,6 +185,13 @@ class Dispatchers {
         // Turn off queuing to allow for normal background execution mode
         queueInitialTasks.value = false
 
+        // This must happen after `queueInitialTasks.set(false)` is run, or it
+        // would be added to a full task queue and be silently dropped.
+        if preInitTaskCount > Constants.maxQueueSize {
+            GleanMetrics.GleanError.preinitTasksOverflow.add(preInitTaskCount)
+        }
+        preInitTaskCount = 0
```
Comment (brizental) [inline glean-core/ios/Glean/Dispatchers.swift]: Just out of curiosity, why do you need zero it out here again?

**Comment by brizental** [comment]
> @brizental wins the reviewer roulette! I can walk you through any of this you may have questions about, if that makes it more comfortable for you to review.

Thanks @travis79 ! I only had one question, the rest was pretty straight forward.

File: glean-core/ios/Glean/Dispatchers.swift (hunk 1 above)
Comment (travis79) [reply glean-core/ios/Glean/Dispatchers.swift]: Mostly for tests which don't necessarily destroy the copy of the `MetricsPingScheduler` held by Glean.  I'll add a comment to help clarify why it was done.

**Pull request merged**